tools/clang/tools/dxcompiler/dxcompileradapter.h

```diff
@@ -51,6 +51,11 @@ class DxcCompilerAdapter: public IDxcCompiler2
   ULONG STDMETHODCALLTYPE Release() override;
   HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void **ppvObject) override;
 
+  void SetDxcCompiler(IDxcCompiler3* pCompilerImpl)
+  {
+      m_pCompilerImpl = pCompilerImpl;
```

Comment:
should we do something about the references to any previously set m_pCompilerImpl? Depending on when this is done, it could result in extra Releases on the DxcCompilerAdapter object which will be passed to this one possibly erroneously. Maybe verify that there are no references instead?

It doesn't match the form of the other SetDxcCompiler functions because it returns nothing, but I don't see this being called anywhere so I can't tell if that's for good reason

Reply:
Yeah, I think the design here is too specific to the intended use case (an implementation of IDxcCompiler3 that contains an instance of IDxcCompilerAdapter and passes itself as the IDxcCompiler3).

I think its not a good general purpose thing to expose in the dxcapi so I will move it to a private implementation where the use case is more restricted and well defined.